gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/iapi/sql/dictionary/StatementTablePermission.java

```diff
@@ -134,7 +135,8 @@ public void check( LanguageConnectionContext lcc,
 	{
 		DataDictionary dd = lcc.getDataDictionary();
 	
-		if( ! hasPermissionOnTable( dd, authorizationId, forGrant))
+		if( ! hasPermissionOnTable( dd, authorizationId, forGrant) && !isSelectOnHiveMetastore
+				(getTableDescriptor(dd)))
```

Comment:
I went with the existing the indentation of these files. If we follow the indentation policy of spaces here, I think, it may not look better.